src/TableHeaderColumn.js

```diff
@@ -100,29 +124,77 @@ class TableHeaderColumn extends Component {
         </span>
       );
     }
-    let sortCaret = sort ? Util.renderReactSortCaret(sort) : defaultCaret;
+    let sortCaret = sort && sort !== 'none' ? Util.renderReactSortCaret(sort) : defaultCaret;
     if (caretRender) {
       sortCaret = caretRender(sort, dataField);
     }
     const classes = classSet(
       typeof className === 'function' ? className() : className,
-      dataSort ? 'sort-column' : '');
+      dataSort ? 'sort-column' : '',
+      'resizable');
 
     const title = typeof children === 'string' ? { title: children } : null;
     return (
       <th ref='header-col'
           className={ classes }
           style={ thStyle }
-          onClick={ this.handleColumnClick }
           { ...title }>
-        { children }{ sortCaret }
+        <div onClick={ this.handleColumnClick } style={ { display: 'inline' } }>{ children }{ sortCaret }</div>
```

Comment:
@allenfang oh this is a change from the internal project i use this table. We also have a column reordering feature so we cant use <th> for this. I will remove this change.